c7n/resources/dynamodb.py

```diff
@@ -229,15 +237,128 @@ def process(self, resources):
+    """
+
+    valid_status = ('ACTIVE',)
+    schema = type_schema('backup',
+                prefix={'type': 'string'})
+    permissions = ('dynamodb:CreateBackup',)
+
+    def process(self, resources):
+        resources = self.filter_table_state(
+            resources, self.valid_status)
+        if not len(resources):
+            return
+
+        c = local_session(self.manager.session_factory).client('dynamodb')
+        futures = {}
+
+        prefix = self.data.get('prefix', 'Backup')
+
+        with self.executor_factory(max_workers=2) as w:
+            for t in resources:
+                futures[w.submit(
+                    c.create_backup,
+                    BackupName=snapshot_identifier(
+                        prefix, t['TableName']),
+                    TableName=t['TableName'])] = t
+            for f in as_completed(futures):
+                t = futures[f]
+                if f.exception():
+                    self.manager.log.warning(
+                        "Could not complete DynamoDB backup table:%s", t)
+                arn = f.result()['BackupDetails']['BackupArn']
+                t['c7n:BackupArn'] = arn
+
+
+@resources.register('dynamodb-backup')
+class Backup(query.QueryResourceManager):
+
+    class resource_type(object):
+        service = 'dynamodb'
+        type = 'table'
+        enum_spec = ('list_backups', 'BackupSummaries', None)
+        detail_spec = None
+        id = 'Table'
+        filter_name = None
+        name = 'TableName'
+        date = 'BackupCreationDateTime'
+        dimension = 'TableName'
+        config_type = 'AWS::DynamoDB::Table'
+
+
+@Backup.action_registry.register('delete')
+class DeleteBackup(BaseAction, StatusFilter):
+    """Deletes backups of a DynamoDB table
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: dynamodb-delete-backup
+                resource: dynamodb-backup
+                filters:
+                  - type: age
+                    days: 28
+                    op: ge
+                actions:
+                  - type: delete
+    """
+
+    valid_status = ('AVAILABLE',)
+    schema = type_schema('delete')
+    permissions = ('dynamodb:DeleteBackup',)
+
+    def process(self, backups):
+        backups = self.filter_backup_state(
+            backups, self.valid_status)
+        if not len(backups):
+            return
+
+        c = local_session(self.manager.session_factory).client('dynamodb')
+
+        for table_set in chunks(backups, 20):
+            self.process_dynamodb_backups(table_set, c)
+
+    def process_dynamodb_backups(self, table_set, c):
+
+        for t in table_set:
+            try:
+                c.delete_backup(
+                    BackupArn=t['BackupArn'])
+            except ClientError as e:
+                if e.response['Error']['Code'] == 'ResourceNotFoundException':
+                    self.log.warning("Could not complete DynamoDB backup table:%s", t)
+                return
```

Comment:
this should be raise instead of return, ie. unknown error should raise exception, instead of passing silently.

Reply:
I've committed a correction for this one.